graphql_compiler/compiler/emit_sql.py

```diff
@@ -112,3 +125,154 @@ def _get_output_columns(node, context):
+        node: SqlNode, the SqlNode the expression applies to.
+        context: CompilationContext, global compilation state and metadata.
+
+    Returns:
+        Expression, SqlAlchemy expression equivalent to the passed compiler expression.
+    """
+    _expression_transformers = {
+        expressions.LocalField: _transform_local_field_to_expression,
+        expressions.Variable: _transform_variable_to_expression,
+        expressions.Literal: _transform_literal_to_expression,
+        expressions.BinaryComposition: _transform_binary_composition_to_expression,
+    }
+    expression_type = type(expression)
+    if expression_type not in _expression_transformers:
+        raise NotImplementedError(
+            u'Unsupported compiler expression "{}" of type "{}" cannot be converted to SQL '
+            u'expression.'.format(expression, type(expression)))
+    return _expression_transformers[expression_type](expression, node, context)
+
+
+def _transform_binary_composition_to_expression(expression, node, context):
+    """Transform a BinaryComposition compiler expression into a SQLAlchemy expression.
+
+    Recursively calls _expression_to_sql to convert its left and right sub-expressions.
+
+    Args:
+        expression: expression, BinaryComposition compiler expression.
+        node: SqlNode, the SqlNode the expression applies to.
+        context: CompilationContext, global compilation state and metadata.
+
+    Returns:
+        Expression, SQLAlchemy expression.
+    """
+    if expression.operator in constants.UNSUPPORTED_OPERATOR_NAMES:
+        raise NotImplementedError(
+            u'Filter operation "{}" is not supported by the SQL backend.'.format(
+                expression.operator))
+    sql_operator = constants.SUPPORTED_OPERATORS[expression.operator]
+    left = _expression_to_sql(expression.left, node, context)
+    right = _expression_to_sql(expression.right, node, context)
+    if sql_operator.cardinality == constants.Cardinality.UNARY:
+        # ensure the operator is grabbed from the Column object
+        if not isinstance(left, Column) and isinstance(right, Column):
+            left, right = right, left
+        clause = getattr(left, sql_operator.name)(right)
+        return clause
+    elif sql_operator.cardinality == constants.Cardinality.BINARY:
+        clause = getattr(sql_expressions, sql_operator.name)(left, right)
+        return clause
+    elif sql_operator.cardinality == constants.Cardinality.LIST_VALUED:
+        if isinstance(left, BindParameter) and isinstance(right, Column):
+            left, right = right, left
+        if not isinstance(right, BindParameter):
+            raise AssertionError(
+                u'List valued operator expects column as left side of expression')
+        if not isinstance(left, Column):
+            raise AssertionError(
+                u'List valued operator expects bind parameter as right side of expression')
```

Comment:
I think the error messages and checks are backwards relative to each other. Also, let's refer to types by their actual names: `BindParameter`, `Column`, etc.